tx/types/types.proto

```diff
@@ -0,0 +1,150 @@
+
+// Tx is the standard type used for broadcasting transactions
+message Tx {
+    // body is the processable content of the transaction
+    TxBody body = 1;
+
+    // auth_info is the authorization related content of the transaction, specifically
+    // signers, signer modes and fee
+    AuthInfo auth_info = 2;
+
+    // signatures are the raw binary signatures of signers specified by body and auth_info
+    repeated bytes signatures = 3;
+}
+
+// SignDoc is the standard type used for signing transaction in SIGN_MODE_DIRECT
+message SignDoc {
+    // body is the TxBody from Tx
+    TxBody body = 1;
+
+    // auth_info is the AuthInfo from Tx
+    AuthInfo auth_info = 2;
+
+    // chain_id is the unique identifier of the chain this transaction targets.
+    // It prevents signed transactions from being used on another chain by an
+    // attacker
+    string chain_id = 3;
+
+    // account_number is the account number of the account in state
+    uint64 account_number = 4;
+
+    // account_sequence starts at 1 rather than 0 to avoid the case where
+    // the default 0 value must be omitted in protobuf serialization
+    uint64 account_sequence = 5;
+}
+
+// TxBody is the body of a transaction that all signers sign over
+message TxBody {
+    // messages are the processable content of the transaction
+    repeated google.protobuf.Any messages = 1;
+
+    // memo is any arbitrary memo to be added to the transaction
+    string memo = 2;
+
+    // timeout is the block height after which this transaction will not
+    // be processed by the chain
+    int64 timeout_height = 3;
+
+    // extension_options are arbitrary options that can be added by chains
+    // when the default options are not sufficient. If any of these are present
+    // and can't be handled, the transaction will be rejected
+    repeated google.protobuf.Any extension_options = 1023;
+
+    // extension_options are arbitrary options that can be added by chains
+    // when the default options are not sufficient. If any of these are present
+    // and can't be handled, they will be ignored
+    repeated google.protobuf.Any non_critical_extension_options = 2047;
+}
+
+// AuthInfo describes the fee and signer modes that are used to sign a transaction
+message AuthInfo {
+    // signer_infos is the list of signer infos which corresponds with
+    // Tx.signatures and expected signers derived from TxBody.messages. All signers
+    // are expected to occur in the same order in each of these locations
+    repeated SignerInfo signer_infos = 1;
+
+    // Fee is the fee and gas limit for the transaction. The first signer is the
+    // primary signer and the one which pays the fee
+    Fee fee = 2;
+}
+
+// SignerInfo describes the public key and signing mode of a single top-level signer
+message SignerInfo {
+    // public_key is the public key of the signer. It is optional for accounts
+    // that already exist in state
+    google.protobuf.Any public_key = 1;
```

Comment:
I'm personally in favor of defaulting to `oneOf` unless there's a specific reason for Any. I know we're using Any in lots of places elsewhere in this spec, but those are all for multichain client UX, which I understand. Does the same argument apply for this? Wouldn't public key be something a bit more standardized?

Reply:
Or since there's a separate file where public key proto definitions are created, shouldn't this use that type directly (instead of either oneOf of Any) ?

Reply:
I was planning to, just not in this PR (see #6217)